File: 5.implementation/redsun/src/main/webapp/static/js/services/PlanningModesService.js


/**
 * Controller for ${entity.name}
 **/


app.factory('planningModeService', function($http, $rootScope) {
	return {
	
		// Insert.
		add: function(planningMode) {
			var urlServer = $rootScope.makePostURL("/planningmode/insert");
			return $http.post(urlServer, planningMode);
		}
		
		// Update.
		, update: function(planningModeId, planningMode) {
			var urlServer = $rootScope.makePostURL("/planningmode/update/" + planningModeId);
			return $http.put(urlServer, planningMode);
		}
		
		// Delete.
		, delete: function(planningModeId) {
			var urlServer = $rootScope.makePostURL("/planningmode/delete/" + planningModeId);
			return $http.delete(urlServer);
		}
		
		// Get by Id.
		, getById: function(planningModeId) {
			var url = $rootScope.makeGetURL("/planningmode/getbyid/" + planningModeId);
			return $http.get(url);
		}
		
		// List for page and filter.
		, getsForPageAndFilter: function(itemsPerPage, pageNo, planningMode) {
			var url = $rootScope.makeGetURL("/planningmode/list/page/filter/" + itemsPerPage + "/" + pageNo);
			return $http.get(url, { params: planningMode });
		}
		
	}// return.
});

File: 5.implementation/redsun-service/src/main/java/com/redsun/service/service/PlanningModesService.java
package com.redsun.service.service;

import com.redsun.service.entities.PlanningModes;
import com.redsun.service.entities.Result;

/**
 * PlanningModes Service interface
 */
public interface PlanningModesService {
	
    // Insert.
	Result insert(final PlanningModes planningModes) ;

    // Update.
	Result update(final PlanningModes planningModes) ;

    // Delete.
	Result delete(final Integer id) ;

	// Get By Id.
	Result getById(final Integer id) ;

    // Exists.
	Result exists(final Integer id) ;

    // Count.
	Result count() ;

	// List for page and filter.
	Result listPlanningModesForPageAndFilter(final int itemsPerPage, final int pageNo, final PlanningModes planningModes);

}
